kbs/src/plugins/implementations/splitapi/manager.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2025 by IBM Corporation
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    fs,
+    path::PathBuf,
+    sync::atomic::{AtomicBool, Ordering},
+    sync::Arc,
+};
+use tempfile::tempdir_in;
+use tokio::sync::RwLock;
+
+use super::backend::{SandboxParams, SplitAPIBackend};
+use super::generator::{CertificateDetails, CredentialGenerator};
+
+/// Credentials (keys and certs for CA, server, and client)
+/// ncessary for the SplitAPI work
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Credentials {
+    pub ca_crt: Vec<u8>,
+    pub client_key: Vec<u8>,
+    pub client_crt: Vec<u8>,
+    pub server_key: Vec<u8>,
+    pub server_crt: Vec<u8>,
+}
+
+/// Credentials necessary for SplitAPI proxy server
+#[derive(Debug, serde::Serialize)]
+pub struct ServerCredentials {
+    pub key: Vec<u8>,
+    pub crt: Vec<u8>,
+    pub ca_crt: Vec<u8>,
+}
+
+/// Manages the credentials generation, handling requests
+/// from backend, and credentials persistence storage
+pub struct CertManager {
+    pub plugin_dir: PathBuf,
+    pub certificate_details: CertificateDetails,
+    pub credblob_file: PathBuf, //String,
+    pub state: Arc<RwLock<HashMap<String, Credentials>>>,
+    credential_loaded_from_file: AtomicBool,
+}
+
+impl CertManager {
+    pub fn new(
+        plugin_dir: PathBuf,
+        blob_file: String,
+        cert_details: &CertificateDetails,
+    ) -> anyhow::Result<Self> {
+        if !plugin_dir.exists() {
+            fs::create_dir_all(&plugin_dir)?;
+            log::info!("plugin dir created = {}", plugin_dir.display());
+        }
+
+        let cblob_file = plugin_dir.as_path().join(blob_file);
+
+        // Initialize the credential manager
+        Ok(Self {
+            plugin_dir,
+            certificate_details: cert_details.clone(),
+            credblob_file: cblob_file,
+            state: Arc::new(RwLock::new(HashMap::new())),
+            credential_loaded_from_file: AtomicBool::new(false),
+        })
+    }
+
+    async fn load_credentials(&self, key: &str) -> Option<Credentials> {
```

Comment:
nit: you are alternating between `credentials` (plural) and `credential` singular in your method names. Probably best to pick one.

Also, this function should probably return `Result<Option<Credentials>>` given that you could have a failure to load the credentials (i.e. if the file doesn't exist or something) or you could legitimately return no credentials.